parsl/tests/test_failure/test_rand_fail.py

```diff
@@ -0,0 +1,171 @@
+            print(e)
+            print("*" * 20)
+            count += 1
+
+    print("Caught failures of  {0}/{1}".format(count, len(fus)))
+
+
+def test_fail_sequence(numtasks=10):
+    ''' Test failure in a sequence of dependencies
+
+    App1 -> App2 ... -> AppN
+    '''
+
+    sleep_dur = 0.1
+    fail_prob = 0.4
+
+    fus = {0: None}
+    for i in range(0, numtasks):
+        print("Chaining {0} to {1}".format(i + 1, fus[i]))
+        fus[i + 1] = sleep_fail(sleep_dur, 0, fail_prob, inputs=[fus[i]])
+
+    # time.sleep(numtasks*sleep_dur)
+    for k in sorted(fus.keys()):
+        try:
+            x = fus[i].result()
+            print("{0} : {1}".format(k, x))
+        except Exception as e:
+            print("{0} : {1}".format(k, e))
+
+    return
+
+
+def test_deps(numtasks=10):
+    ''' Random failures in branches of Map -> Map -> reduce
+
+    App1   App2  ... AppN
+    '''
+
+    fus = []
+    for i in range(0, numtasks):
+        fu = sleep_fail(0.2, 0, .4)
+        fus.extend([fu])
+
+    '''
+    App1   App2  ... AppN
+    |       |        |
+    V       V        V
+    App1   App2  ... AppN
+    '''
+
+    fus_2 = []
+    for fu in fus:
+        fu = sleep_fail(0, 0, .8, inputs=[fu])
+        fus_2.extend([fu])
+
+    '''
+    App1   App2  ... AppN
+      |       |        |
+      V       V        V
+    App1   App2  ... AppN
+       \      |       /
+        \     |      /
+          App_Final
+    '''
+    fu_final = sleep_fail(1, 0, 0, inputs=fus_2)
+
+    try:
+        print("Final status : ", fu_final.result())
+    except parsl.dataflow.error.DependencyError as e:
+        print("Caught the right exception")
+        print("Exception : ", e)
+    except Exception as e:
+        assert 5 == 1, "Expected DependencyError got : %s" % e
+    else:
+        print("Shoot! no errors ")
+
+
+@App('python', dfk)
+def sleep_then_fail(sleep_dur=0.1):
+    import time
+    import math
+    time.sleep(sleep_dur)
+    math.ceil("Trigger TypeError")
+    return 0
+
+
+def test_fail_nowait(numtasks=10):
+    ''' Test basic error handling, with no dependent failures
+    '''
+    import time
+    fus = []
+    for i in range(0, numtasks):
+        fu = sleep_then_fail(sleep_dur=0.1)
+        fus.extend([fu])
+
+    try:
+        [x.result() for x in fus]
+    except Exception as e:
+        assert isinstance(
+            e, TypeError), "Expected a TypeError, got {}".format(e)
+
+    # fus[0].result()
+    time.sleep(1)
+    print("Done")
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-c", "--count", default="10",
+                        help="Count of apps to launch")
+    parser.add_argument("-d", "--debug", action='store_true',
+                        help="Count of apps to launch")
+    args = parser.parse_args()
+
+    if args.debug:
+        set_stream_logger()
+
+    test_fail_nowait(numtasks=int(args.count))
+    # test_no_deps(numtasks=int(args.count))
```

Comment:
Code bits in the main are only for testing by hand. This is pretty non-consequential

Reply:
@yadudoc Maybe we should split up tests that are meant to only be used by hand and tests that will eventually automatically be run when we set up Jenkins? If these are only meant to be used by hand, it is wasted processing to have them run automatically.

Reply:
To clarify, I mean that if one is developing, you want to be able to run `nosetests parsl/tests` and have it tell you if you broke something as quickly as possible, rather than having whoever runs nosetests need to know which tests are meant to be run by hand or not.